src/Widgets/TerminalView.vala

```diff
@@ -257,6 +263,44 @@ public class Terminal.TerminalView : Granite.Bin {
         open_in_new_window_action.set_enabled (page != null && tab_view.n_pages > 1);
     }
 
+    private bool key_pressed (uint keyval, uint keycode, Gdk.ModifierType modifiers) {
+        switch (keyval) {
+            case Gdk.Key.@1: //alt+[1-8]
+            case Gdk.Key.@2:
+            case Gdk.Key.@3:
+            case Gdk.Key.@4:
+            case Gdk.Key.@5:
+            case Gdk.Key.@6:
+            case Gdk.Key.@7:
+            case Gdk.Key.@8:
+                if (ALT_MASK in modifiers &&
```

Comment:
We could probably determine tab_index first and then test `n_pages > tab_index`? Not sure its worth having two `if` clauses.